pkgs/applications/video/animdl/default.nix

```diff
@@ -0,0 +1,70 @@
+{ python3Packages,
+  buildPythonApplication,
+  fetchFromGitHub,
+  lib,
+  poetry-core,
+  anchor-kr,
+  anitopy,
+  click,
+  cssselect,
+  httpx,
+  lxml,
+  packaging,
+  pkginfo,
+  pycryptodomex,
+  pyyaml,
+  regex,
+  rich,
+  tqdm,
+  yarl
+}:
+buildPythonApplication rec {
+  pname = "animdl";
+  version = "1.7.27";
+
+  src = fetchFromGitHub {
+    owner = "justfoolingaround";
+    repo = "animdl";
+    # Using the commit hash because upstream does not have releases. https://github.com/justfoolingaround/animdl/issues/277
+    rev = "c7c3b79198e66695e0bbbc576f9d9b788616957f";
+    hash = "sha256-/XPVWBitFYsUkb9WMlR5F2amAPcVnDtJUgFXP2gXyNk=";
+  };
+
+  format = "pyproject";
+  nativeBuildInputs = [
+    poetry-core
+  ];
+  propagatedBuildInputs = [
+    anchor-kr
+    anitopy
+    click
+    cssselect
+    httpx
+    lxml
+    packaging
+    pkginfo
+    pycryptodomex
+    pyyaml
+    regex
+    rich
+    tqdm
+    yarl
+  ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'pycryptodomex = "~=3.14.1"' 'pycryptodomex = "*"' \
+      --replace 'rich = ">=13.3.1,<13.3.4"' 'rich = "*"' \
+      --replace 'version = "4.9.1"' 'version = "*"' \
+      --replace 'comtypes = "~=1.1.11"' ' '
+  '';
```

Comment:
we have a hook called `pythonRelaxDepsHook` which you should use instead of this. grep through nixpkgs for examples